torch/csrc/utils/python_arg_parser.h

```diff
@@ -123,6 +124,19 @@ inline at::Scalar PythonArgs::scalar(int i) {
   return at::Scalar(static_cast<int64_t>(THPUtils_unpackLong(args[i])));
 }
 
+inline std::vector<at::Tensor> PythonArgs::tensorlist(int i) {
+  if (!args[i]) return std::vector<at::Tensor>();
+  PyObject* arg = args[i];
+  auto tuple = PyTuple_Check(arg);
+  auto size = tuple ? PyTuple_GET_SIZE(arg) : PyList_GET_SIZE(arg);
+  std::vector<at::Tensor> res(size);
+  for (int idx = 0; idx < size; idx++) {
+    PyObject* obj = tuple ? PyTuple_GET_ITEM(arg, idx) : PyList_GET_ITEM(arg, idx);
+    res[idx] = reinterpret_cast<THPVariable*>(obj)->cdata;
+  }
+  return res;
```

Comment:
There's no error checking here, which is worrying for me. It is checked somewhere else, right?

Reply:
I assume you are referring to the variable type checking?  I don't see that being done anywhere else.  I think both this and the existing PythonArgs::tensor above should be calling THPVariable_Check.

Reply:
Type checks are done during parse.

Reply:
Just talked with @colesbury about this; the parse code does enough checking to figure out which signature to dispatch to, but not full checking.  In this particular case, it just matches a TensorList if the python object is a tuple or list, so we actually do need to add the checking as suggested.